### PR TITLE
Fix #1200: NullPointerException in BuildImageConfiguration

### LINKED ISSUE #1200: NullPointerException after upgrading to 0.29.0
### Description
After upgrading from 0.28.0 to 0.29.0 we get the folloing NullPointerException when trying to build a docker image:
```
Caused by: java.lang.NullPointerException
    at io.fabric8.maven.docker.config.BuildImageConfiguration.getAbsoluteContextDirPath (BuildImageConfiguration.java:329)
    at io.fabric8.maven.docker.assembly.DockerAssemblyManager$1.customize (DockerAssemblyManager.java:133)
    at io.fabric8.maven.docker.assembly.DockerAssemblyManager.createBuildTarBall (DockerAssemblyManager.java:312)
    at io.fabric8.maven.docker.assembly.DockerAssemblyManager.createDockerTarArchive (DockerAssemblyManager.java:179)
    at io.fabric8.maven.docker.service.ArchiveService.createArchive (ArchiveService.java:123)
    at io.fabric8.maven.docker.service.ArchiveService.createArchive (ArchiveService.java:118)
    at io.fabric8.maven.docker.service.BuildService.buildImage (BuildService.java:117)
    at io.fabric8.maven.docker.service.BuildService.buildImage (BuildService.java:65)
    at io.fabric8.maven.docker.BuildMojo.buildAndTag (BuildMojo.java:73)
    at io.fabric8.maven.docker.BuildMojo.processImageConfig (BuildMojo.java:104)
    at io.fabric8.maven.docker.BuildMojo.executeInternal (BuildMojo.java:60)
    at io.fabric8.maven.docker.AbstractDockerMojo.execute (AbstractDockerMojo.java:225)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
```

Other than the version, we did not change any configuration.

### Info

* d-m-p version : 0.29.0
* Maven version (`mvn -v`) :
```
Apache Maven 3.5.4
Maven home: /usr/share/maven
Java version: 11.0.1, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-11-openjdk-amd64
Default locale: en_GB, platform encoding: UTF-8
OS name: "linux", version: "4.18.0-17-generic", arch: "amd64", family: "unix"
```
* Docker version : Docker version 18.06.1-ce, build e68fc7a

### PR DESCRIPTION
**Pull request by rohanKanojia**
Fix #1200 

Fixed my mistake. This is based upon suspicion seeing the stack trace provided by the user. I am still not able to reproduce this exception.

### CODE REVIEW

**Comment by rhuss** [comment]
Thanks ! Let's wait until we get more information about the actual error to reproduce it properly.

File: src/main/java/io/fabric8/maven/docker/config/BuildImageConfiguration.java [hunk 1]
```diff
@@ -162,7 +162,11 @@ public boolean isDockerFileMode() {
     }
 
     public File getContextDir() {
-        return contextDir != null ? new File(contextDir) : getDockerFile().getParentFile();
+        return contextDir != null
+                ? new File(contextDir)
+                : (getDockerFile().getParentFile() != null
+                     ? getDockerFile().getParentFile()
+                     : new File(Paths.get("").toAbsolutePath().toString()));
```
Comment (rhuss) [inline src/main/java/io/fabric8/maven/docker/config/BuildImageConfiguration.java]: I mean, it should be sufficient to just return ".". It gets converted to an absolute path later.

Comment (rohanKanojia) [reply src/main/java/io/fabric8/maven/docker/config/BuildImageConfiguration.java]: but method is returning a `File`, you mean returning a `new File(".")`?

Comment (rhuss) [reply src/main/java/io/fabric8/maven/docker/config/BuildImageConfiguration.java]: Yeah, that was my idea. But not 100% sure.

File: src/main/java/io/fabric8/maven/docker/config/BuildImageConfiguration.java [hunk 2]
```diff
@@ -330,7 +330,16 @@ public File getAbsoluteContextDirPath(MojoParameters mojoParams) {
     }
 
     public File getAbsoluteDockerFilePath(MojoParameters mojoParams) {
-        return EnvUtil.prepareAbsoluteSourceDirPath(mojoParams, getDockerFile().getPath());
+        File absoluteDockerFilePath = EnvUtil.prepareAbsoluteSourceDirPath(mojoParams, getDockerFile().getPath());
```
Comment (rohanKanojia) [inline src/main/java/io/fabric8/maven/docker/config/BuildImageConfiguration.java]: Is it okay to update `contextDir` in a getter method like this??

Or shall we change initAndValidate method in order to handle Dockerfile early also, but not sure that has access to MojoParams

**Pull request merged**